demos/classification_benchmark_demo/cpp_gapi/main.cpp

```diff
@@ -0,0 +1,549 @@
+            }.cfgNumRequests(config.maxAsyncRequests)
+             .pluginConfig(config.getLegacyConfig());
+        // clang-format on
+
+        auto kernels = cv::gapi::kernels<custom::OCVLocateROI, custom::OCVTopK>();
+        auto pipeline = comp.compileStreaming(cv::compile_args(kernels, cv::gapi::networks(net)));
+
+        /** Output container for result **/
+        cv::Mat output;
+        IndexScore infer_result;
+
+        /** ---------------- The execution part ---------------- **/
+        cap = openImagesCapture(FLAGS_i,
+                                FLAGS_loop,
+                                read_type::safe,
+                                0,
+                                std::numeric_limits<size_t>::max(),
+                                stringToSize(FLAGS_res));
+        cv::gapi::wip::IStreamSource::Ptr media_cap;
+        if (FLAGS_use_onevpl) {
+            auto onevpl_params = util::parseVPLParams(FLAGS_onevpl_params);
+            if (FLAGS_onevpl_pool_size != 0) {
+                onevpl_params.push_back(
+                    cv::gapi::wip::onevpl::CfgParam::create_frames_pool_size(FLAGS_onevpl_pool_size));
+            }
+            media_cap = cv::gapi::wip::make_onevpl_src(FLAGS_i, std::move(onevpl_params));
+        } else {
+            media_cap = cv::gapi::wip::make_src<custom::MediaCommonCapSrc>(cap);
+        }
+
+        auto pipeline_inputs = cv::gin(std::move(media_cap));
+        pipeline.setSource(std::move(pipeline_inputs));
+        std::string windowName = "Classification Benchmark demo G-API";
+        int delay = 1;
+
+        Presenter presenter(FLAGS_u, 0);
+        int width;
+        int height;
+        std::vector<std::string> gridMatRowsCols = split(FLAGS_res, 'x');
+        if (gridMatRowsCols.size() != 2) {
+            throw std::runtime_error("The value of ClassificationGridMat resolution flag is not valid.");
+        } else {
+            width = std::stoi(gridMatRowsCols[0]);
+            height = std::stoi(gridMatRowsCols[1]);
+        }
+
+        ClassificationGridMat gridMat(presenter, cv::Size(width, height));
+        bool keepRunning = true;
+        size_t framesNum = 0;
+        long long correctPredictionsCount = 0;
+        unsigned int framesNumOnCalculationStart = 0;
+        bool isStart = true;
+        double accuracy = 0;
+        bool isTestMode = true;
+        std::chrono::steady_clock::duration elapsedSeconds = std::chrono::steady_clock::duration(0);
+        std::chrono::seconds testDuration = std::chrono::seconds(3);
+        std::chrono::seconds fpsCalculationDuration = std::chrono::seconds(1);
+        auto startTime = std::chrono::steady_clock::now();
+        pipeline.start();
+        IndexScore::LabelsStorage top_k_scored_labels;
+        top_k_scored_labels.reserve(FLAGS_nt);
+        int64_t timestamp = 0;
+        while (keepRunning && elapsedSeconds < std::chrono::seconds(FLAGS_time) && pipeline.pull(cv::gout(output, infer_result, timestamp))) {
```

Comment:
Is there a reason you suggest to introduce `-loop` instead of `-time` from cpp demo? I prefer to keep demos aligned, unless you provide other arguments

Reply:
cannot assigned in `true` by default, as it declared in common gapi header.
So overridden in benchmark main.cpp

Reply:
done